lib/matplotlib/backends/backend_wx.py

```diff
@@ -717,11 +718,14 @@ def _get_key(self, event):
         else:
             key = None
 
-        for meth, prefix in (
-                [event.AltDown, 'alt'],
-                [event.ControlDown, 'ctrl'], ):
-            if meth():
-                key = '{0}+{1}'.format(prefix, key)
+        for meth, prefix, key_name in (
+                [event.ControlDown, 'ctrl', 'control'],
+                [event.AltDown, 'alt', 'alt'],
+                [event.ShiftDown, 'shift', 'shift'],):
+            if meth() and not key_name == key:
+                if not (key_name == 'shift' and key.isupper()):
```

Comment:
This looks like it just gives up on non-ascii (?!)  base on the `if...elif...else` just above this?  Am I reading that right?

I think fixing that is out of scope for this PR (to keep it from getting to big, small PRs are easier to review and get in).

Reply:
```python
key = chr(keyval)
# wx always returns an uppercase, so make it lowercase if the shift
# key is not depressed (NOTE: this will not handle Caps Lock)
if not event.ShiftDown():
    key = key.lower()
```
Thanks to this fragment of code, we get a lowercase letter if shift is not pressed. If it's, then we get an uppercase letter.
Then, the following line of code makes sure we don't get `shift+A` and rather get `shift`, enter and `A`.
```python
if not (key_name == 'shift' and key.isupper()):
```
How ever we are still interested on getting `shift+2` or `shift+]` as Wx doesn't translate the keys. Hence only when shift was pressed and the key is an uppercase letter we are interested in not adding the shift prefix. If we don't do this, then there would be one more inconsistency on Wx backend.